### PR TITLE
main terraform/docker code changes for payments.web.ch.gov.uk

### PR DESCRIPTION
**Pull request by nghessen-ch**
JU-726 and JU-725 main terraform/docker code changes for payments.web.ch.gov.uk

### CODE REVIEW

File: terraform/groups/ecs-service/variables.tf [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  default     = "SSM_VERSION_"
+}
+
+variable "use_set_environment_files" {
+  type        = bool
+  default     = false
+  description = "Toggle default global and shared environment files"
+}
+
+variable "payments_web_ch_gov_uk_version" {
```
Comment (chillier-ch) [inline terraform/groups/ecs-service/variables.tf:114]: This is fine but might be nicer just to name it as payments_web_version

File: terraform/groups/ecs-service/data.tf [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+data "aws_ecs_cluster" "ecs_cluster" {
+  cluster_name = "${local.name_prefix}-cluster"
+}
+
+data "aws_iam_role" "ecs_cluster_iam_role" {
+  name = "${local.name_prefix}-ecs-task-execution-role"
+}
+
+data "aws_lb" "service_lb" {
+  name = "${var.environment}-chs-chgovuk"
```
Comment (chillier-ch) [inline terraform/groups/ecs-service/data.tf:37]: It looks like this service may have it's own loadbalancer: https://github.com/companieshouse/ch-service-terraform/blob/master/aws-mm-elb/module-chs/elb-payments-web.tf

File: README.md [hunk 3]
```diff
@@ -40,4 +40,33 @@ command on the command line:
...
+### What does this code do?
+
+The code present in this repository is used to define and deploy a dockerised container in AWS ECS.
+This is done by calling a [module](https://github.com/companieshouse/terraform-modules/tree/main/aws/ecs) from terraform-modules. Application specific attributes are injected and the service is then deployed using Terraform via the CICD platform 'Concourse'.
+
+
+Application specific attributes | Value                                | Description
+:---------|:-----------------------------------------------------------------------------|:-----------
+**ECS Cluster**        |payments-service                                     | ECS cluster (stack) the service belongs to
+**Load balancer**      |{env}-chs-chgovuk                                            | The load balancer that sits in front of the service
```
Comment (chillier-ch) [inline README.md:56]: May need to update this based on lb defined in data.tf

**Comment by nghessen-ch** [comment]
```

  # module.secrets.aws_ssm_parameter.secret_parameters["chs_payment_privilege_api_key"] will be created
  + resource "aws_ssm_parameter" "secret_parameters" {
      + arn            = (known after apply)
      + data_type      = (known after apply)
      + description    = "chs_payment_privilege_api_key"
      + id             = (known after apply)
      + insecure_value = (known after apply)
      + key_id         = "3890acb0-0dc1-4927-975f-a2492279174b"
      + name           = "/payments-web-cidev/chs_payment_privilege_api_key"
      + overwrite      = true
      + tags_all       = (known after apply)
      + tier           = (known after apply)
      + type           = "SecureString"
      + value          = (sensitive value)
      + version        = (known after apply)
    }

Plan: 15 to add, 0 to change, 0 to destroy.
```

**Review by chillier-ch — APPROVED** [review]

**Pull request merged**